src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -135,43 +247,167 @@ public void run() {
 
     @Override
     public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
-        if (endpoint.getInputCluster(ZclColorControlCluster.CLUSTER_ID) == null
+        clusterColorControl = (ZclColorControlCluster) endpoint.getInputCluster(ZclColorControlCluster.CLUSTER_ID);
+
+        if (clusterColorControl == null || endpoint.getInputCluster(ZclOnOffCluster.CLUSTER_ID) == null
                 || endpoint.getInputCluster(ZclLevelControlCluster.CLUSTER_ID) == null) {
             return null;
         }
+
+        try {
+            if(!clusterColorControl.discoverAttributes(false).get()) {
+                logger.warn("{}: Cannot determine whether device endpoint supports RGB color. Assuming it does by now (checking again later)", endpoint.getEndpointAddress());
+            }
+            else if(clusterColorControl.getSupportedAttributes().contains(ZclColorControlCluster.ATTR_CURRENTHUE)) {
+                logger.debug("{}: Device endpoint supports HUE/SATURATION color", endpoint.getEndpointAddress());
+            }
+            else if(clusterColorControl.getSupportedAttributes().contains(ZclColorControlCluster.ATTR_CURRENTX)) {
+                logger.debug("{}: Device endpoint supports XY color", endpoint.getEndpointAddress());
+            }
+            else {
+                logger.debug("{}: Device endpoint does not support RGB color (probably only color temperature)", endpoint.getEndpointAddress());
+                return null;
+            }
+        }
+        catch(InterruptedException | ExecutionException e) {
+                logger.warn("{}: Exception checking whether device endpoint supports RGB color. Assuming it does by now (checking again later)", endpoint.getEndpointAddress(), e);
+        }
+
         return createChannel(thingUID, endpoint, ZigBeeBindingConstants.CHANNEL_COLOR_COLOR,
                 ZigBeeBindingConstants.ITEM_TYPE_COLOR, "Color");
     }
 
+    protected void updateOnOff(OnOffType onoff) {
+        HSBType oldHSB = currentHSB;
+        updateChannelState(onoff);
+        logger.debug("{}: OnOff updated. {} -> {}", endpoint.getEndpointAddress(), oldHSB.toString(), onoff.toString());
+    }
+
+    protected void updateBrightness(PercentType brightness) {
+        HSBType oldHSB = currentHSB;
+        HSBType newHSB = new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), brightness);
+        currentHSB = newHSB;
+        updateChannelState(brightness);
+        updateChannelState(newHSB);
+        logger.debug("{}: Brightness updated. {} -> {}", endpoint.getEndpointAddress(), oldHSB.toString(), newHSB.toString());
```

Comment:
I'd suggest to remove this log and add one in either the updateChannelState method, or the setChannelState method so there's a consistent log entry.

Reply:
Will do